docs/bundler/migration.md

```diff
@@ -0,0 +1,1107 @@
+
+---
+
+- `jsxFragment`
+- `jsxFragment`
+- Not supported in JS API, configure in `tsconfig.json`
+
+---
+
+- `jsxImportSource`
+- `jsxImportSource`
+- Not supported in JS API, configure in `tsconfig.json`
+
+---
+
+- `jsxSideEffects`
+- `jsxSideEffects`
+- Not supported in JS API, configure in `tsconfig.json`
+
+---
+
+- `keepNames`
+- n/a
+- Not supported
+
+---
+
+- `legalComments`
+- n/a
+- Not supported
+
+---
+
+- `loader`
+- n/a
+- Not supported in JS API
+
+---
+
+- `logLevel`
+- n/a
+- Not supported
+
+---
+
+- `logLimit`
+- n/a
+- Not supported
+
+---
+
+- `logOverride`
+- n/a
+- Not supported
+
+---
+
+- `mainFields`
+- n/a
+- Not supported
+
+---
+
+- `mangleCache`
+- n/a
+- Not supported
+
+---
+
+- `mangleProps`
+- n/a
+- Not supported
+
+---
+
+- `mangleQuoted`
+- n/a
+- Not supported
+
+---
+
+- `metafile`
+- `manifest`
+- When `manifest` is `true`, the result of `Bun.build()` will contain a `manifest` property. The manifest is compatible with esbuild's metafile format.
+
+---
+
+- `minify`
+- `minify`
+- In Bun, `minify` can be a boolean or an object.
+
+  ```ts
+  Bun.build({
+    entrypoints: ['./index.tsx'],
+    // enable all minification
+    minify: true
+
+    // granular options
+    minify: {
+      identifiers: true,
+      syntax: true,
+      whitespace: true
+    }
+  })
+  ```
+
+---
+
+- `minifyIdentifiers`
+- `minify.identifiers`
+- Nested object syntax
+
+---
+
+- `minifySyntax`
+- `minify.syntax`
+- Nested object syntax
+
+---
+
+- `minifyWhitespace`
+- `minify.whitespace`
+- Nested object syntax
+
+---
+
+- `nodePaths`
+- n/a
+- Not supported
+
+---
+
+- `outExtension`
+- n/a
+- Not supported
+
+---
+
+- `outbase`
+- `root`
+- Different name
+
+---
+
+- `outdir`
+- `outdir`
+- No differences
+
+---
+
+- `outfile`
+- `outfile`
+- No differences
+
+---
+
+- `packages`
+- n/a
+- Not supported, use `external`
+
+---
+
+- `platform`
+- `target`
+- Supports `"bun"`, `"node"`, and `"browser"` (the default)
+
+---
+
+- `plugins`
+- `plugins`
+- Bun's plugin API is esbuild-compatible
+
+---
+
+- `preserveSymlinks`
+- n/a
+- Not supported
+
+---
+
+- `publicPath`
+- `publicPath`
+- No differences
+
+---
+
+- `pure`
+- n/a
+- Not supported
+
+---
+
+- `reserveProps`
+- n/a
+- Not supported
+
+---
+
+- `resolveExtensions`
+- n/a
+- Not supported
+
+---
+
+- `sourceRoot`
+- n/a
+- Not supported
+
+---
+
+- `sourcemap`
+- `sourcemap`
+- Supports `"inline"`, `"external"`, and `"none"`
+
+---
+
+- `sourcesContent`
+- n/a
+- Not supported
+
+---
+
+- `splitting`
+- `splitting`
+- No differences
+
+---
+
+- `stdin`
+- n/a
+- Not supported
+
+---
+
+- `supported`
+- n/a
+- Not supported
+
+---
+
+- `target`
+- n/a
+- No support for syntax downleveling
+
+---
+
+- `treeShaking`
+- n/a
+- Always `true`
+
+---
+
+- `tsconfig`
+- n/a
+- Not supported
+
+---
+
+- `write`
+- n/a
+- Set to `true` if `outdir`/`outfile` is set, otherwise `false`
+
+---
+
+{% /table %}
+
+## Plugin API
```

Comment:
we should have a table of the list of callback types we support, as we support onLoad and onResolve, we do not have

- `onStart`
- `onEnd`
- `onDispose`

we also do not have these properties on that `build` object
- `initialOptions`, which is slightly problematic as for esbuild compat we need to match esbuild
- `esbuild` which is a copy of the esbuild library itself. we cannot easily copy that
- `resolve` a function to run the resolver. this we could implement later